src/admin/class-wordlift-admin-entity-list.php

```diff
@@ -80,4 +80,73 @@ public function render_custom_columns( $column, $entity_id ) {
 		}
 		
 	}
+	
+	/**
+	 * Add 4W select box before the 'Filter' button.
+	 *
+	 * @since 3.3.0
+	 */
+	public function add_4W_filter() {
+		global $typenow;
+		
+		// Only show on entity list page
+		if( $typenow !== Wordlift_Entity_Service::TYPE_NAME ){
+			return;
+		}
+		
+		// Was a W already selected?
+		$selected = isset( $_GET['4W_filter'] ) ? $_GET['4W_filter'] : '' ;
+		
+		// Build select box with the 4W
+		$all_w = array(
+			'Select W',
+			WL_WHAT_RELATION,
+			WL_WHO_RELATION,
+			WL_WHERE_RELATION,
+			WL_WHEN_RELATION
+		);
+		$output  = '<select name="4W_filter" id="dropdown_4W_type">';
+		foreach ( $all_w as $w ) {
+			$default  = ( $selected == $w ) ? 'selected' : ''; 
+			$output  .= "<option value='$w' $default >" . ucfirst( __( $w, 'wordlift' ) ) . '</option>';
+		}
+		$output .= '</select>';
+		
+		// Print on page
+		echo $output;
+	}
+
+	/**
+	 * Server side response operations for the 4W filter set in *add_4W_filter*
+	 *
+	 * @since 3.3.0
+	 * 
+	 * @param array $clauses WP main query clauses.
+	 * 
+	 * @return array Modified clauses.
+	 */
+	public function add_4W_filter_query( $clauses ) {
+				
+		global $typenow, $wpdb;
+		
+		// Only apply on entity list page and if the 4W_filter query param is set
+		if( ! ( $typenow == Wordlift_Entity_Service::TYPE_NAME && isset( $_GET['4W_filter'] ) ) ) {
+			return $clauses;
+		}
+			
+		// Check a valid W was requested
+		$requested_w = $_GET['4W_filter'];
+		if ( ! in_array( $requested_w, array( WL_WHAT_RELATION, WL_WHO_RELATION, WL_WHERE_RELATION, WL_WHEN_RELATION )) ) {
+			return $clauses;
+		}
+		
+		$w_table = wl_core_get_relation_instances_table_name();
+		
+		// Change WP main query clauses
+		$clauses['join']     .= "INNER JOIN $w_table ON $wpdb->posts.ID = $w_table.object_id";
+		$clauses['where']    .= "AND $w_table.predicate = '$requested_w'";
```

Comment:
I have difficulties using prepare:
```
WordPress database error: [You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near ''wp_wl_relation_instances' ON 'wp_posts'.ID = 'wp_wl_relation_instances'.object_' at line 1]
SELECT SQL_CALC_FOUND_ROWS DISTINCT wp_posts.ID FROM wp_posts INNER JOIN 'wp_wl_relation_instances' ON 'wp_posts'.ID = 'wp_wl_relation_instances'.object_id WHERE 1=1 AND wp_posts.post_type = 'entity' AND (wp_posts.post_status = 'publish' OR wp_posts.post_status = 'future' OR wp_posts.post_status = 'draft' OR wp_posts.post_status = 'pending' OR wp_posts.post_status = 'private')AND wp_wl_relation_instances.predicate = 'what' ORDER BY wp_posts.post_date DESC LIMIT 0, 20
```
Seems mandatory to be used with user supplied params, but not with table names:
http://wordpress.stackexchange.com/questions/93830/really-simple-query-giving-error-in-sql-syntax